app/uk/gov/hmrc/hmrcemailrenderer/templates/tdq/params/ValidConnectionMethodBase64EncodedDetails.scala

```diff
@@ -41,9 +56,38 @@ final case class HeadersValidation(
   headerOrHeaders: String,
   errors: Set[Problem],
   warnings: Set[Problem]
-)
+) {
 
-final case class Problem(message: String, percentage: Int) {
+  val prettyHeaderOrHeaders: String = headerOrHeaders.split(',').map(_.trim).map(prettyHeader).mkString(", ")
+
+  private def prettyHeader(header: String): String = header match {
```

Comment:
Can this conversion be pushed into notifier or reporter? Reporter already has a `FraudPreventionHeader` class that has `name` (the display name version) and `canonicalName` (the all-lowercase version), which is used as part of the metrics functionality.

Reply:
I did that and unfortunately this makes the code much more complicated as these are also saved as emails so you end up with `Web application via server` as the method value in the DB. The same for the header values.